docs/html/classcachemere_1_1policy_1_1EvictionLRU.html

```diff
@@ -0,0 +1,297 @@
+  </table>
+  </dd>
+</dl>
+
+</div>
+</div>
+<a id="a8925052de1b9e8318b2f8e49f78ee120"></a>
+<h2 class="memtitle"><span class="permalink"><a href="#a8925052de1b9e8318b2f8e49f78ee120">&#9670;&nbsp;</a></span>on_update()</h2>
+
+<div class="memitem">
+<div class="memproto">
+<div class="memtemplate">
+template&lt;typename Key , typename Value &gt; </div>
+      <table class="memname">
+        <tr>
+          <td class="memname">void <a class="el" href="classcachemere_1_1policy_1_1EvictionLRU.html">cachemere::policy::EvictionLRU</a>&lt; Key, Value &gt;::on_update </td>
+          <td>(</td>
+          <td class="paramtype">const <a class="el" href="structcachemere_1_1detail_1_1Item.html">CacheItem</a> &amp;&#160;</td>
+          <td class="paramname"><em>item</em></td><td>)</td>
+          <td></td>
+        </tr>
+      </table>
+</div><div class="memdoc">
+
+<p>Update event handler. </p>
+<p>Moves the provided item to the front of the list. </p><dl class="params"><dt>Parameters</dt><dd>
+  <table class="params">
+    <tr><td class="paramname">item</td><td>The item that has been updated in the cache. </td></tr>
+  </table>
+  </dd>
+</dl>
+
+</div>
+</div>
+<a id="a0f106be385324e1cc5bd312c340bf971"></a>
+<h2 class="memtitle"><span class="permalink"><a href="#a0f106be385324e1cc5bd312c340bf971">&#9670;&nbsp;</a></span>victim_begin()</h2>
+
+<div class="memitem">
+<div class="memproto">
+<div class="memtemplate">
+template&lt;typename Key , typename Value &gt; </div>
+      <table class="memname">
+        <tr>
+          <td class="memname"><a class="el" href="classcachemere_1_1policy_1_1EvictionLRU_1_1VictimIterator.html">VictimIterator</a> <a class="el" href="classcachemere_1_1policy_1_1EvictionLRU.html">cachemere::policy::EvictionLRU</a>&lt; Key, Value &gt;::victim_begin </td>
+          <td>(</td>
+          <td class="paramname"></td><td>)</td>
+          <td> const</td>
+        </tr>
+      </table>
+</div><div class="memdoc">
+
+<p>Get an iterator to the first item that should be evicted. </p>
+<dl class="section return"><dt>Returns</dt><dd>An item iterator. </dd></dl>
+
+</div>
+</div>
+<a id="ac42ec1741f4df21088889a5c02b4c21b"></a>
+<h2 class="memtitle"><span class="permalink"><a href="#ac42ec1741f4df21088889a5c02b4c21b">&#9670;&nbsp;</a></span>victim_end()</h2>
+
+<div class="memitem">
+<div class="memproto">
+<div class="memtemplate">
+template&lt;typename Key , typename Value &gt; </div>
+      <table class="memname">
+        <tr>
+          <td class="memname"><a class="el" href="classcachemere_1_1policy_1_1EvictionLRU_1_1VictimIterator.html">VictimIterator</a> <a class="el" href="classcachemere_1_1policy_1_1EvictionLRU.html">cachemere::policy::EvictionLRU</a>&lt; Key, Value &gt;::victim_end </td>
+          <td>(</td>
+          <td class="paramname"></td><td>)</td>
+          <td> const</td>
+        </tr>
+      </table>
+</div><div class="memdoc">
+
+<p>Get an end iterator. </p>
```

Comment:
An [end iterator](https://www.cplusplus.com/reference/iterator/end/) is an iterator pointing _one past the end_ of the sequence. I'd consider this to be a well-known concept for most C++ developers (as most containers in the standard library conform to this API), although a link couldn't hurt :smile: 

As for 
> It always returns the last element that would become the victim?

I think something is not quite clear in the docs.
The LRU and Segmented LRU policies internally store keys from hottest to coldest. This means that the last item in the list is the one that will be evicted.

However the policy interface states that `victim_begin` must yield an iterator to _the first item that should be evicted_, so effectively, in the case of LRU-based policies, the iterator returned will walk the internal structure in reverse order.
I think should be clarified by adding a comment at the class level for both LRU policies that indicates that the policy internally stores keys from hottest to coldest, as well as a comment on `victim_begin` stating that the victim iterator will walk the internal structure backwards - what do you think?

Reply:
ah that makes sense
And yes, I think quick notes would be useful for readers to clarify the directions of the data structures, because it can be counterintuitive when you first look at it :)